src/ccl/charts/line-graph.component.ts

```diff
@@ -0,0 +1,119 @@
+import { Component, ViewEncapsulation, ElementRef } from '@angular/core';
+import * as D3 from 'd3';
+
+/*
+ * Line graph component
+ * Contains all logic for drawing a line graph
+ */
+
+@Component({
+  selector: 'line-graph',
+  encapsulation: ViewEncapsulation.None,
+  template: `<ng-content></ng-content>`,
+  inputs: [ 'data', 'indicator' ],
+  styleUrls: [ '../../assets/css/chart.css']
+})
+
+export class LineGraph {
+
+  public data: Object;
```

Comment:
ere's an example of doing it this way, I also have limited experience here so there might be a better way:
Define model:
https://gitlab.com/CloudNiner/disc.us/blob/feature/webapp/ng2/src/app/models/Discussion.ts
Retrieve data from server as instance of model:
https://gitlab.com/CloudNiner/disc.us/blob/feature/webapp/ng2/src/app/services/nwsoffice.service.ts#L19
Use data in component:
https://gitlab.com/CloudNiner/disc.us/blob/feature/webapp/ng2/src/app/office-detail/office-detail.component.ts

If you need to do a transformation in between pulling the data and providing to a component, that might be a bit more involved. Could probably have a function that returns the target type, and just creates an object and then typecasts it to start.